debian/control

```diff
@@ -5,7 +5,8 @@ Source: resistance-miner
 Section: misc
 Priority: optional
 Maintainer: Patrick Schleizer <adrelanos@riseup.net>
-Build-Depends: debhelper (>= 12), genmkfile
+Build-Depends: debhelper (>= 12), genmkfile, pkg-config,
+ libcurl4-openssl-dev
```

Comment:
Why not list autoconf, automake, make, gcc as build dependencies?

More importantly, where are we running the actual build - invoke ./autogen.sh and so on?

Reply:
> Why not list autoconf, automake, make, gcc as build dependencies?

Because these are part of `build-essential` and shall not be added to debian/control as per Debian convention (which is supervised by lintian, a static lintian package checking tool). The build works inside a clean chroot using cowbuilder. So there's probably no missing build dependencies otherwise the build would abort.

> More importantly, where are we running the actual build - invoke ./autogen.sh and so on?

Debhelper does that. Except:

```
dh build
   dh_update_autotools_config
   dh_autoreconf
        find ! -ipath "./debian/*" -a ! \( -path '*/.git/*' -o -path '*/.hg/*' -o -path '*/.bzr/*' -o -path '*/.svn/*' -o -path '*/CVS/*' \) -a  -type f -exec md5sum {} + -o -type l -printf "symlink  %p
" > debian/autoreconf.before
        grep -q ^XDT_ configure.ac
        autoreconf -f -i
```

Debhelper automates a ton of things. ./autogen.sh, configure, make is basic debhelper.